### PR TITLE
support for profiles

### LINKED ISSUE #299: Default command line options
I'd like a file where I can put my default command line options that any run of `cucumber-js` in that directory will use.

Example if I have a file `.cucumber-js` with the content
```
--strict --format pretty
```
Then anytime I run cucumber-js in that directory those options are automatically applied and I can override them if need be.

### LINKED ISSUE #329: Make pretty the default formatter on CLI
@aslakhellesoy thanks for raising this.

### PR DESCRIPTION
**Pull request by charlierudolph**
@jbpros 

fixes #299

### CODE REVIEW

File: lib/cucumber/cli/argument_parser.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-function ArgumentParser(argv) {
+function ArgumentParser(argv, defaultaArgv) {
```
Comment (a-morales) [inline lib/cucumber/cli/argument_parser.js]: typo : `defaultArgv`

**Comment by charlierudolph** [comment]
@jbpros any thoughts?

**Comment by jbpros** [comment]
Sorry for the late reply. This is great and needs to be in for ages.

I wonder whether we need something as fancy as profiles and [cucumber.yml](https://github.com/cucumber/cucumber/wiki/cucumber.yml) or simply the ability to specify alternate configuration files like the one you propose.

**Comment by jbpros** [comment]
Also, what about the file name? 

Can be

    .cucumber-js
    .cucumberjs
    .cucumber.js
    Cucumberjsfile
    cucumberjs.conf
    ...

And how does Windows behave with dot files nowadays?

**Comment by jbpros** [comment]
Original explanation from @charlierudolph on #299:

I'd like a file where I can put my default command line options that any run of `cucumber-js` in that directory will use.

Example if I have a file `.cucumber-js` with the content
```
--strict --format pretty
```
Then anytime I run cucumber-js in that directory those options are automatically applied and I can override them if need be.

**Comment by charlierudolph** [comment]
I'm fine implementing profiles, just wanted to get something in to get started. If we do profiles how I'd suggest we just make it `cucumber.yml`. Good with just about any filename you listed aside from  `Cucumberjsfile`. Don't know about Windows.

**Comment by jbpros** [comment]
I don't think we can use `cucumber.yml`, there are projects using both Cucumber.js and Cucumber-ruby out there (Cucumber.js itself does that). 

I quite like `.cucumber.js` and I think it should be a plain JS object.

It'd be nice if you can give a try at profiles. Let's not worry about preprocessing of the file like cucumber.yml's ERB parsing.

WDYT?

PS: I think `.Xxxfile` files are stupid, I was just teasing with that suggestion :)

**Comment by charlierudolph** [comment]
Okay so a normal javascript file then?
```js
module.exports = {
  default: 'options'
  other_profile: 'options'
}
```
I don't think the leading dot is necessary in the filename. How about just `cucumber.js`?

And implement profiles as described [here](https://github.com/cucumber/cucumber/wiki/cucumber.yml)? (aside from ERB and autotest integration)

**Comment by charlierudolph** [comment]
@jbpros got the feature test passing if you'd like to take a look. Still need to update lint, and add/update unit tests.

**Comment by charlierudolph** [comment]
@jbpros this is ready to go

**Comment by charlierudolph** [comment]
@jbpros any thoughts?

**Comment by jbpros** [comment]
@charlierudolph thank you, this is awesome!

Merged in as 188db52563ebedc255340f44eee84cf7ee2cf1b4

**Pull request closed without merging**